### PR TITLE
Optimized consensus signature share verification

### PR DESCRIPTION
**Pull request by iulianpascalau**
- optimized consensus signature share verification

### CODE REVIEW

File: consensus/spos/bls/subroundSignature.go [hunk 1]
```diff
@@ -158,6 +158,11 @@ func (sr *subroundSignature) receivedSignature(cnsDta *consensus.Message) bool {
 	}
 
 	currentMultiSigner := sr.MultiSigner()
+	err = currentMultiSigner.VerifySignatureShare(uint16(index), cnsDta.SignatureShare, sr.GetData(), nil)
+	if err != nil {
+		return false
```
Comment (SebastianMarian) [inline consensus/spos/bls/subroundSignature.go:163]: You can log the error,  just to know which is the error when this method returns false in a critical situation and to keep the pattern used in this method:

		log.Debug("receivedSignature.VerifySignatureShare",
			"index", index,
			"error", err.Error())

Comment (iulianpascalau) [reply consensus/spos/bls/subroundSignature.go:167]: done

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
image

**Review by raduchis — APPROVED** [review]
Tested in production and endround time was decreased from ~600 ms to ~100 ms

**Pull request merged**